Cargo.toml

```diff
@@ -20,9 +20,10 @@ once_cell = "1"
 regex ={version = "1", default-features = false, features = ["std"]}
 tantivy-fst = "0.3"
 memmap = {version = "0.7", optional=true}
+lz4_flex = { version = "0.7", default-features = false, features = ["checked-decode"], optional=true }
 lz4 = {version="1", optional=true}
 brotli = {version="3.3.0", optional=true}
-snap = "1"
+snap = {version="1.0", optional=true}
```

Comment:
version=1

We are a library, so we do not want to impose a fixed version. By stopping at the major we ensure compatibility (assuming the maintainer follows SemVer), we allow users to rely on more recent versions of snap, and we reduce the chance that they might end up with two+ versions of snap in their binary application.

Before 1., the x in version 0.x.y, is treated as a major release which is why you see a lot of of 0.x in this. 

In applications on the other hand, we rely on Cargo.lock to fix versions.

Reply:
That makes sense. I was a little confused, about brotli being specific.